pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -284,20 +291,6 @@ func balanceDomains(
 		// also (just for tracking), add them to the list of pods in the lower topology
 		aboveToEvict := sortedDomains[j].pods[len(sortedDomains[j].pods)-movePods:]
 		for k := range aboveToEvict {
-			// if the pod has a hard nodeAffinity or nodeSelector that only matches this node,
```

Comment:
> // don't bother evicting it as it will just end up back on the same node
> // however we still account for it "being evicted" so the algorithm can complete

The code still needs to stay here based on the comment. I will keep this for @damemi

Reply:
Sorry, trying to remember exactly what I meant by this comment. I think removing the check here is okay. Pretty sure that `we still account for it "being evicted"` is actually referring to this line: https://github.com/kubernetes-sigs/descheduler/blob/24c0ca2/pkg/descheduler/strategies/topologyspreadconstraint.go#L294

Where all the pods in `aboveToEvict` are removed from the in-memory representation of `sortedDomains[j]`, whether or not they were actually added to `podsForEviction`.

tl;dr, we won't need this affinity/taints check if `evictable.IsEvictable` does it before actually evicting them. But it will result in more calls to `IsEvictable`, because some of the pods in `podsForEviction` aren't being skipped without it.

Reply:
So, I think it's just that the comment was in a bad spot